pkg/build/api/types.go

```diff
@@ -328,6 +328,9 @@ type GitBuildSource struct {
 
 	// HTTPSProxy is a proxy used to reach the git repository over https
 	HTTPSProxy *string
+
+	// URLCheckTimeoutSeconds is the timeout in seconds to use when checking connectivity to the git server
+	URLCheckTimeoutSeconds *int
```

Comment:
I don't see a point in this check having a distinct timeout from the build
timeout.  If you're willing to wait 30 minutes for a build, why aren't you
willing to wait 30m for git ls remote?

On Mon, May 23, 2016 at 3:45 PM, Cesar Wong <notifications@github.com>
wrote:

> In pkg/build/api/types.go
> <https://github.com/openshift/origin/pull/8990#discussion_r64276299>:
>
> > @@ -328,6 +328,9 @@ type GitBuildSource struct {
> >
> >  	// HTTPSProxy is a proxy used to reach the git repository over https
> >  	HTTPSProxy *string
> > +
> > +	// URLCheckTimeoutSeconds is the timeout in seconds to use when checking connectivity to the git server
> > +	URLCheckTimeoutSeconds *int
>
> @deads2k <https://github.com/deads2k> I don't know we can apply it across
> the board to all repository types. This is a very git-specific check (git
> ls-remote)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8990/files/09988784f42909e7b7cd9619c04c27904d62d89d#r64276299>
>